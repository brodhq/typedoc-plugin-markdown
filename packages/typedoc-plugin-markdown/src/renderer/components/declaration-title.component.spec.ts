import { TestApp } from '../../../test/test-app';
import { DeclarationTitleComponent } from './declaration-title.component';
describe(`Declaration Title Component`, () => {
  let testApp: TestApp;

  beforeAll(async () => {
    testApp = new TestApp(['declarations.ts']);
    await testApp.bootstrap();
  });

  test(`should compile a const with default value`, () => {
    const model = testApp.findReflectionByName('constWithDefaultValue');
    expect(DeclarationTitleComponent(model)).toMatchSnapshot();
  });

  test(`should compile a let with default value`, () => {
    const model = testApp.findReflectionByName('letWithDefaultValue');
    expect(DeclarationTitleComponent(model)).toMatchSnapshot();
  });

  test(`should compile an undefined declaration`, () => {
    const model = testApp.findReflectionByName('undefinedNumberDeclaration');
    expect(DeclarationTitleComponent(model)).toMatchSnapshot();
  });
});
