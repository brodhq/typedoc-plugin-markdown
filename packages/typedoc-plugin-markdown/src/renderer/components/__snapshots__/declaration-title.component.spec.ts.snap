// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Declaration Title Component should compile a const with default value 1`] = `"• **constWithDefaultValue**: \`\`\\"<foo>\\"\`\`"`;

exports[`Declaration Title Component should compile a let with default value 1`] = `"• **letWithDefaultValue**: \`string\` = \`'hello'\`"`;

exports[`Declaration Title Component should compile an undefined declaration 1`] = `"• **undefinedNumberDeclaration**: \`number\`"`;
